_posts/2015-04-01-aurelia-io.md

```diff
@@ -0,0 +1,87 @@
+
+### What is Aurelia? ###
+
+Aurelia is a framework that is relatively new and not many have heard of it - or used it. It's not stable for production use yet - but it is looking pretty good already. It's also looking a lot like Angular 2, just much better. This is of course no big surprise, since one of the key developers is Rob Eisenberg, previously a member of Angular 2 team.
+
+### What's cool about Aurelia? ###
+
+What fascinates me about Aurelia? Many things. One core thing is that it's based on using ECMAScript 6 - and 7. This means you are able to code with future languages - right now. But it still works in todays browsers - thanks to Babel 6to5 transpiler, that forms a core part of this framework. This means support for all current Evergreen browsers, and there's even some work being done for IE10 and IE9 support. It's beautiful, and applicable today.
+
+Don't believe me? Take a look at this:
+
+{% highlight javascript %}
+import {HttpClient} from 'aurelia-http-client';
+import _ from 'underscore';
+
+export class App {
+
+  static inject() { return [HttpClient]; }
+
+  constructor(http){
+        this.title = 'Whatever';
+        this.items = [];
+        this.http = http;
+  }
+
+  // ... more good stuff
+}
+{% endhighlight %}
+
+What is happening here? A lot. Aurelia uses ES6 module system so import becomes much simpler than with Angular+RequireJS combination. We see dependency injection in action, we provide some metadata and HttpClient becomes injected in our constructor. And no more $scope! Our scope is now the viewmodel, and we simply store any interesting variables in this (that now works better, too)
+
+### How's the view? ###
+
+Aurelia uses view-viewmodel architecture, and what we see here is viewmodel named App. The view for this viewmodel would be named app.html, and it looks like this:
+
+{% highlight html %}
+<template>
+    <section>
+        <h2>${title}</h2>
+
+        <input value.bind="query">
+
+        <div repeat.for="item of items">
+                    ${item.name} ${item.created_at}
+        </div>
+    </section>
+</template>
+</code>
+{% endhighlight %}
+
+Here we can see example of another powerful Aurelia feature - binding. There's a whole binding engine at work to figure out what's the best way to implement the binding based on your code. There are several options. Any html attribute may be bound, in this case you can see repeat.for, but similarly you can bind with value.bind="somevalue", or href.bind="somehref" - you can also use one-way binds or two-way binds by replacing '.bind' part with '.one-way', or '.two-way', to be more precise.
+
+To me it seems like Aurelia is very clean and pure code, compared to the mess that Angular 1 and 2 very easily tend to become. Using strong points of ES6 is definitely a plus, since ECMAScript 6 will come, one day. Being able to already start making transition and learning to apply it will be valuable.
```

Comment:
Ehdotus vikan lauseen lyhentämiseksi: Starting the transition now will be valuable for me.